nupic/research/frameworks/vernon/mixins/multi_cycle_lr.py

```diff
@@ -43,10 +43,10 @@ def setup_experiment(self, config):
 
         ignored_class = config.pop("lr_scheduler_class", None)
         if ignored_class is not None and ignored_class != OneCycleLR:
-            self.logger.warning("Ignoring lr_scheduler_class, using OneCycleLR")
+            logging.warning("Ignoring lr_scheduler_class, using OneCycleLR")
```

Comment:
These need to be `self.logger` because the Distributed logic often disables loggers: https://github.com/numenta/nupic.research/blob/3fd4124bba28094884c42bb29fc07114b1cf3440/nupic/research/frameworks/vernon/distributed/experiments/components/distributed_base.py#L89

Reply:
I may be able help here. This was a temporary fix we did to get the code running. A bug came out where MultiCycleLR mixin references self.logger before calling setup_experiment(), which is where the logger is initialized. How did you get around that?

Reply:
Ah, interesting. Then we can move these lines to after the call to `super().setup_experiment(config)`. In general our policy will need to be "don't log before calling `super().setup_experiment`".

Reply:
Are you suggesting moving the entire block or just the logging lines? There are some changes to the config prior to calling setup_experiment, I understand that moving them after setup_experiment will change the intended behavior.

Reply:
Just the logging lines.